app/controllers/admin/claims_controller.rb

```diff
@@ -22,14 +22,12 @@ def show
   end
 
   def search
-    return unless params[:reference].present?
+    return unless params[:query].present?
 
-    claim = Claim.find_by(reference: params[:reference].upcase)
+    @claims = Claim::Search.new(params[:query]).claims
```

Comment:
Let's avoid an n+1 and preload eligibilities (used by the `Claim#policy` method):
```suggestion
    @claims = Claim::Search.new(params[:query]).claims.includes(:eligibility)
```